### PR TITLE
Fix nested element with inner extended simple attribute using JDBC multivalue join

### PR DESCRIPTION
**Pull request by fernandor777**
This PR fixes an issue with nested element names and extended simpleContent elements and attributes involved when using a JDBC multivalue join, making Hale export a non existent feature type.

### CODE REVIEW

File: io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingTest.java [hunk 1]
```diff
@@ -416,8 +422,8 @@ public void testReferenceTypeJoinHandler() throws Exception {
 
 		processJoinAlignment(alignment, chainingConf);
 
-		logMapping(mappingWrapper.getMainMapping());
-		logMapping(mappingWrapper.getIncludedTypesMapping());
+//		logMapping(mappingWrapper.getMainMapping());
+//		logMapping(mappingWrapper.getIncludedTypesMapping());
```
Comment (nmco) [inline io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingTest.java]: ?

File: io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
```
Comment (nmco) [inline io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandlerTest.java]: Don't use this header.

File: io/plugins/it.geosolutions.hale.io.appschema/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandler.java [hunk 3]
```diff
@@ -187,15 +195,20 @@ public AttributeMappingType handlePropertyTransformation(Cell typeCell, Cell pro
 					// don't translate mapping, will do it (or have done it)
 					// elsewhere!
 					featureType = null;
+					if (LOG.isDebugEnabled())
+						LOG.debug(
+								"Don't map properties that belong to a feature chaining "
+										+ "configuration other than the current one. \n"
+										+ "chainConf = {} \n" + "isSameParentType = {}",
+								chainConf, isSameParentType);
```
Comment (nmco) [inline io/plugins/it.geosolutions.hale.io.appschema/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandler.java:203]: Strange formatting.

File: io/plugins/it.geosolutions.hale.io.appschema/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandler.java [hunk 4]
```diff
@@ -607,7 +637,7 @@ private void setEncodeIfEmpty(ClientProperty clientProperty) {
 				attributeMapping.setEncodeIfEmpty(true);
 			}
 		} catch (CQLException e) {
-			log.warn("Cannot set encodeIfEmpty value. Reason: " + e.getMessage());
+			LOG.warn("Cannot set encodeIfEmpty value. Reason: " + e.getMessage());
```
Comment (nmco) [inline io/plugins/it.geosolutions.hale.io.appschema/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandler.java]: The message should say that the exception \ situation is ignore and skipped, unless that's not the intention.

**Review by nmco — APPROVED** [review]
Looks good to me, adds tests and improves logging, just some minor fixes required.

File: io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingTest.java (hunk 1 above)
Comment (fernandor777) [reply io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingTest.java]: Oops, a leftover.  Fixing it.

File: io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandlerTest.java (hunk 2 above)
Comment (fernandor777) [reply io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/internal/AbstractPropertyTransformationHandlerTest.java]: Fixing.

**Pull request merged**